javascript-engine/src/index.ts

```diff
@@ -0,0 +1,99 @@
+import init, { Engine } from '@unleash/yggdrasil-wasm'
+import { checkResponse, readResponse } from './utils'
+import { Strategies, type Strategy } from './strategies'
+import type {
+  Context,
+  FeatureDefinition,
+  MetricsBucket,
+  State,
+  Variant
+} from './types'
+export * from './types'
+
+await init()
+
+export const DISABLED_VARIANT: Variant = {
+  name: 'disabled',
+  enabled: false,
+  featureEnabled: false
+}
+
+export class UnleashEngine {
+  private engine: Engine
+  private strategies: Strategies
+
+  constructor(strategies?: Strategy[]) {
+    this.engine = new Engine()
+
+    const knownStrategiesResponse = this.engine.builtInStrategies()
+    const knownStrategies = readResponse<string[]>(knownStrategiesResponse)
+
+    this.strategies = new Strategies(knownStrategies)
+
+    if (strategies != null) {
+      this.strategies.registerCustomStrategies(strategies)
+    }
+  }
+
+  public takeState(state: State): void {
+    const response = this.engine.takeState(state)
+    checkResponse(response)
+
+    this.strategies.mapFeatures(state)
+  }
+
+  public isEnabled(toggleName: string, context: Context): boolean {
+    const customStrategyPayload = this.strategies.getCustomStrategyPayload(
+      toggleName,
+      context
+    )
+
+    const response = this.engine.checkEnabled(
+      toggleName,
+      context,
+      customStrategyPayload
+    )
+
+    return readResponse<boolean>(response) || false
+  }
+
+  public getVariant(toggleName: string, context: Context): Variant {
+    const customStrategyPayload = this.strategies.getCustomStrategyPayload(
+      toggleName,
+      context
+    )
+
+    const response = this.engine.checkVariant(
+      toggleName,
+      context,
+      customStrategyPayload
+    )
+
+    return (
```

Comment:
Same thing here. The SDKs will let you set a fallback variant if the variant is completely missing, I don't think doing a fixed disabled will work here

Reply:
Makes sense, better to address it now. https://github.com/Unleash/yggdrasil/pull/226/commits/dab0c08de5fc5f9e841028e4fdf63350c76da52d

Reply:
Awesome, cheers!